src/index.js

```diff
@@ -1,30 +1,95 @@
 const express = require("express");
 const bodyParser = require("body-parser");
-
+const passport = require("passport");
+const Strategy = require("passport-local").Strategy;
+const db = require("./db");
 const app = express();
 
+// Configure the local strategy for use by Passport.
+passport.use(
+  new Strategy(function (username, password, cb) {
+    db.users.findByUsername(username, function (err, user) {
+      if (err) {
+        return cb(err);
+      }
+      if (!user) {
+        return cb(null, false);
+      }
+      if (user.password != password) {
+        return cb(null, false);
+      }
+      return cb(null, user);
+    });
+  })
+);
+
+// Configure Passport authenticated session persistence.
+passport.serializeUser(function (user, cb) {
+  cb(null, user.id);
+});
+
+passport.deserializeUser(function (id, cb) {
+  db.users.findById(id, function (err, user) {
+    if (err) {
+      return cb(err);
+    }
+    cb(null, user);
+  });
+});
+
+// Configure view engine
+app.set("views", __dirname + "/views");
+app.set("view engine", "ejs");
+
 // middlewares
 // parse application/x-www-form-urlencoded
+app.use(require("morgan")("combined"));
 app.use(bodyParser.urlencoded({ extended: false }));
+app.use(
+  require("express-session")({
+    secret: "keyboard cat",
+    resave: false,
+    saveUninitialized: false,
+  })
+);
+
+//Init Passport
+app.use(passport.initialize());
+app.use(passport.session());
 
 // parse application/json
 app.use(bodyParser.json());
 
 const port = process.env.PORT || 3000;
 
 // configure routes
-app.get("/", (req, res) => {
-  res.send("This is the landing page\nSam was here\nRon was also here");
+app.get("/", function (req, res) {
+  res.render("home", { user: req.user });
 });
 
-app.get("/home", (req, res) => {
-  res.send("This is the home page");
+app.get("/login", function (req, res) {
+  res.render("login");
 });
 
-app.get("/health", (req, res) => {
-  res.send("Healthy!");
+app.post(
+  "/login",
+  passport.authenticate("local", { failureRedirect: "/login" }),
+  function (req, res) {
+    res.redirect("/");
+  }
+);
+
+app.get("/logout", function (req, res) {
+  req.logout();
+  res.redirect("/");
 });
 
+app.get("/profile", require("connect-ensure-login").ensureLoggedIn(), function (
```

Comment:
Requires should come at the top of the file, but AFAICT you don't need this package.

```suggestion
 app.get("/profile", passport.authenticate('local'), function (
```

Reply:
100% agree, this was just some lazy copy paste

Reply:
Moved require to top of file as style dictates. However the suggested change does not work as written and returns a bad request when trying to get profile